crucible-mir/src/Mir/Trans.hs

```diff
@@ -166,27 +166,36 @@ transConstVal _ty (Some (IsizeRepr)) (ConstInt i) =
 -- staticSlicePlace does the first four of these actions; addrOfPlace
 -- does the last two.
 --
-transConstVal (M.TyRef _ _) (Some MirSliceRepr) (M.ConstSliceRef defid len) = do
-    place <- staticSlicePlace len defid
-    addr <- addrOfPlace place
-    return addr
-
-transConstVal _ty (Some (MirVectorRepr u8Repr@(C.BVRepr w))) (M.ConstStrBody bs)
-  | Just Refl <- testEquality w (knownNat @8) = do
+transConstVal (M.TyRef (M.TySlice ty) _) (Some MirSliceRepr) (M.ConstSliceRef defid len) = do
+    Some tpr <- tyToReprM ty
+    place <- staticSlicePlace len (Some tpr) defid
+    addrOfPlace place
+transConstVal (M.TyRef M.TyStr _) (Some MirSliceRepr) (M.ConstSliceRef defid len) = do
+    let tpr = C.BVRepr $ knownNat @8
+    place <- staticSlicePlace len (Some tpr) defid
+    addrOfPlace place
+
+transConstVal _ty (Some MirAggregateRepr) (M.ConstStrBody bs) = do
     let bytes = map (\b -> R.App (eBVLit (knownNat @8) (toInteger b))) (BS.unpack bs)
-    let vec = R.App $ E.VectorLit u8Repr (V.fromList bytes)
-    mirVec <- mirVector_fromVector u8Repr vec
-    return $ MirExp (MirVectorRepr u8Repr) mirVec
+    ag <- mirAggregate_uninit_constSize (fromIntegral $ length bytes)
+    ag' <- foldM
+        (\ag' (i, b) -> mirAggregate_set i 1 knownRepr b ag')
+        ag (zip [0..] bytes)
```

Comment:
Should this mention `TODO: hardcoded size=1` somewhere? (Same question applies to the similar use of `mirAggregate_set` a few lines down.)